lib/screenplay/outfront/sftp.ex

```diff
@@ -11,29 +11,38 @@ defmodule Screenplay.Outfront.SFTP do
   @retries 3
   @sftp_client_module Application.compile_env(:screenplay, :sftp_client_module)
 
-  def set_takeover_images(stations, portrait_png, landscape_png) do
+  def run(work_fn) do
     conn = start_connection()
 
-    Enum.each(stations, fn station ->
-      write_image(conn, station, @portrait_dir, portrait_png)
-      write_image(conn, station, @landscape_dir, landscape_png)
-    end)
+    try do
+      work_fn.(conn)
+    rescue
+      e ->
+        log_error(e)
+    after
+      @sftp_client_module.disconnect(conn)
+    end
+  end
 
-    @sftp_client_module.disconnect(conn)
+  def set_takeover_images(stations, portrait_png, landscape_png) do
+    run(fn conn ->
+      Enum.each(stations, fn station ->
+        write_image(conn, station, @portrait_dir, portrait_png)
+        write_image(conn, station, @landscape_dir, landscape_png)
+      end)
+    end)
   end
 
   def clear_takeover_images(stations) do
-    conn = start_connection()
-
-    Enum.each(
-      stations,
-      fn station ->
-        delete_image(conn, station, @portrait_dir)
-        delete_image(conn, station, @landscape_dir)
-      end
-    )
-
-    @sftp_client_module.disconnect(conn)
+    run(fn conn ->
+      Enum.each(
+        stations,
+        fn station ->
+          delete_image(conn, station, @portrait_dir)
+          delete_image(conn, station, @landscape_dir)
+        end
+      )
+    end)
```

Comment:
Nice, it looks like we're gaining some better cleanup of the connection by wrapping these ✨